src/API/Trace/Propagation/B3SinglePropagator.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\API\Trace\Propagation;
+
+use function count;
+use function explode;
+use OpenTelemetry\API\Trace\AbstractSpan;
+use OpenTelemetry\API\Trace\SpanContext;
+use OpenTelemetry\API\Trace\SpanContextInterface;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\ContextKey;
+use OpenTelemetry\Context\Propagation\ArrayAccessGetterSetter;
+use OpenTelemetry\Context\Propagation\PropagationGetterInterface;
+use OpenTelemetry\Context\Propagation\PropagationSetterInterface;
+use OpenTelemetry\Context\Propagation\TextMapPropagatorInterface;
+
+/**
+ * B3 is a propagator that supports the specification for the B3 single
+ * header "b3" used for trace context propagation across service boundaries.
+ * (https://github.com/openzipkin/b3-propagation#single-header)
+ */
+final class B3SinglePropagator implements TextMapPropagatorInterface
+{
+    public const B3 = 'b3';
+
+    public static ContextKey $B3_DEBUG_FLAG_KEY;
```

Comment:
This context key has to be shared between single header and multi header propagator.

Reply:
Created new class `B3DebugFlagContextKey` as you suggested.